packages/components/src/InputTime/InputTime.tsx

```diff
@@ -52,3 +48,33 @@ export function InputTime({
     }
   }
 }
+
+function dateToTimeString(date?: Date): string {
+  if (date == undefined || !(date instanceof Date)) {
+    return "";
+  }
+
+  // Extract hours and minutes from the Date object
+  const hours = date.getHours().toString().padStart(2, "0");
+  const minutes = date.getMinutes().toString().padStart(2, "0");
+
+  // Return the time string in HH:MM format
+  return `${hours}:${minutes}`;
+}
+
+export function timeStringToDate(timeString: string): Date | undefined {
+  try {
+    const [hours, minutes] = timeString.split(":").map(Number);
+
+    if (isNaN(hours) || isNaN(minutes)) {
```

Comment:
Should we add some validation that the hours and minutes values are valid (ie minutes less than 61) or are we confident that if it is a number string it is a valid one?

Reply:
The current implementation relies heavily on the `useTimePredict` hook which already tightly constrains what values can get set.

That said, I added some bound checking here just in case.